indico/modules/rb/client/js/common/rooms/selectors.js

```diff
@@ -137,3 +137,19 @@ export const getBuildings = createSelector(
     return Array.from(buildings).sort((a, b) => parseInt(a, 10) - parseInt(b, 10));
   }
 );
+
+export const getLocations = createSelector(
+  getAllRooms,
+  allRooms => {
+    const locations = Object.values(allRooms)
+      .map(({locationName, locationId}) => {
+        return {
+          name: locationName,
+          id: locationId,
+        };
+      })
+      .filter((value, index, self) => index === self.findIndex(t => t.id === value.id));
```

Comment:
That looks like an incredibly ineffective way of removing duplicates. It'd be better to replace this whole selector function with one that simply builds a map using `new Map(Object.values(allRooms).map(room => [room.locationId, room.locationName]))` (untested, but should work) - that can then of course be converted to an array of objects if you prefer that over an ES Map object